pkg/kritis/admission/admission.go

```diff
@@ -18,45 +18,149 @@ package admission
 
 import (
 	"encoding/json"
-	"log"
-	"net/http"
-
+	"fmt"
 	"github.com/grafeas/kritis/pkg/kritis/admission/constants"
+	kritisv1beta1 "github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
+	"github.com/grafeas/kritis/pkg/kritis/crd/securitypolicy"
+	"github.com/grafeas/kritis/pkg/kritis/kubectl/plugins/resolve"
+	"github.com/grafeas/kritis/pkg/kritis/metadata"
+	"github.com/grafeas/kritis/pkg/kritis/metadata/containeranalysis"
+	"github.com/grafeas/kritis/pkg/kritis/pods"
+	"io/ioutil"
 	"k8s.io/api/admission/v1beta1"
+	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+	"net/http"
 )
 
-// This is very basic admission controller which allows all pods.
+type config struct {
+	retrievePod           func(r *http.Request) (*v1.Pod, error)
+	metadata              func() (metadata.MetadataFetcher, error)
```

Comment:
can we use the interface here directly, instead of mocking out the function?

Reply:
yah that's what I tried initially, but I think when fetching the metadata client it looks for some google auth so it doesn't work with travis